staff/carlos-calvo/array-functions/fill.2.js

```diff
@@ -0,0 +1,198 @@
+/**
+ * Abstraction of fill.
+ * 
+ * Fills an array from one position to other.
+ * 
+ * @param {Array} array 
+ * @param {*} value 
+ * @param {number} start 
+ * @param {number} end 
+ * 
+ * @throws {Error} - If too many arguments (> 4)
+ * @throws {TypeError} - If array is not an array
+ */
+function fill(array, value, start, end) {
+    if (arguments.length > 4) throw Error('too many arguments');
+
+    if (!(array instanceof Array))
+        throw new TypeError(array + ' is not an array');
+
+    start = start === undefined ? 0 : (start < 0 ? array.length + start : start);
```

Comment:
refactor to ===> start = !start ? 0 : (start < 0 ? array.length + start : start);